src/metadata/metadata_partition_structs.h

```diff
@@ -11,33 +11,38 @@
 #include "../eviction/eviction.h"
 
 struct ocf_user_part_config {
-        char name[OCF_IO_CLASS_NAME_MAX];
-        uint32_t min_size;
-        uint32_t max_size;
-        int16_t priority;
-        ocf_cache_mode_t cache_mode;
-        struct {
-                uint8_t valid : 1;
-                uint8_t added : 1;
-                uint8_t eviction : 1;
-                        /*!< This bits is setting during partition sorting,
-                         * and means that can evict from this partition
-                         */
-        } flags;
+	char name[OCF_IO_CLASS_NAME_MAX];
+	uint32_t min_size;
+	uint32_t max_size;
+	int16_t priority;
+	ocf_cache_mode_t cache_mode;
+	struct {
+		uint8_t valid : 1;
+		uint8_t added : 1;
+		uint8_t eviction : 1;
+			/*!< This bits is setting during partition sorting,
+			* and means that can evict from this partition
+			*/
+	} flags;
 };
 
 struct ocf_user_part_runtime {
-        uint32_t curr_size;
-        uint32_t head;
-        struct eviction_policy eviction;
-        struct cleaning_policy cleaning;
+	env_atomic curr_size;
+
+	/* Number of cache lines reserved for in flight IO */
+	uint32_t reserved_clines_cnt;
+	env_spinlock reserved_clines_lock;
```

Comment:
This is part of persistent metadata, written to disk without any serialization upon management operation. env_atomic and env_spinlock are opaque types, potentially containing pointers. For this reason you should not put atomics/spinlocks in ocf_user_part_runtime. Please either change the type (curr_size back to regular type) or move these members outside this struct. 

Also consider what actually belongs to persistent metadata - spinlock obviously does not, while curr_size probably does (depende on usage details).